src/include/ipc/probe.h

```diff
@@ -0,0 +1,78 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/* Header for data packets sent via compressed PCM from extraction probes */
+struct probe_data_packet {
+	uint32_t sync_word;
+	uint32_t buffer_id;
+	uint32_t format;
+	uint32_t timestamp_low;
+	uint32_t timestamp_high;
+	uint64_t checksum;
+	uint32_t data_size_bytes;
+	uint32_t data[];
+} __attribute__((packed));
+
+struct probe_dma {
+	uint32_t stream_tag;
+	uint32_t dma_buffer_size;
+} __attribute__((packed));
+
+struct probe_point {
+	uint32_t buffer_id;
+	uint32_t purpose;
+	uint32_t stream_tag;
```

Comment:
is this intended to link with a probe_dma?
what is the value on extract, where multiple probe points share the same DMA, hence the same stream_tag?

Reply:
1 extration dma (stream) - N probe points per stream
N injection dmas (streams) - 1 probe point per stream

buffer_id -> module/pin
purpuse -> enum EXTRACT/INJECT
stream_tag -> stream identifier

Reply:
@plbossart In case of extraction, stream_tag passed from user space will be ignored and in getter function, proper extraction stream tag will be returned.